realm/kotlin-extensions/src/main/kotlin/io/realm/kotlin/RealmResultsExtensions.kt

```diff
@@ -63,38 +57,17 @@ import kotlinx.coroutines.flow.flowOf
  */
 @Beta
 fun <T : RealmModel> RealmResults<T>.toFlow(): Flow<RealmResults<T>> {
-    // Return "as is" if frozen, there will be no listening for changes
-    if (realm.isFrozen) {
-        return flowOf(this)
-    }
-
-    val config = realm.configuration
-
-    return callbackFlow {
-        val results = this@toFlow
-
-        // Do nothing if the results are invalid
-        if (!results.isValid) {
-            return@callbackFlow
-        }
-
-        // Get instance to ensure the Realm is open for as long as we are listening
-        val flowRealm = Realm.getInstance(config)
-        val listener = RealmChangeListener<RealmResults<T>> { listenerResults ->
-            offer(listenerResults.freeze())
-        }
-
-        results.addChangeListener(listener)
-
-        // Emit current (frozen) value
-        offer(freeze())
-
-        awaitClose {
-            // Remove listener and cleanup
-            if (!flowRealm.isClosed) {
-                results.removeChangeListener(listener)
-                flowRealm.close()
-            }
-        }
+    @Suppress("INACCESSIBLE_TYPE")
+    return when (val realmInstance = baseRealm) {
+        is Realm -> realmInstance.configuration.flowFactory?.from(baseRealm as Realm, this)
+                ?: throw IllegalStateException("Missing flow factory in Realm configuration.")
+        is DynamicRealm -> realmInstance.configuration.flowFactory?.from(baseRealm as DynamicRealm, this)
+                ?: throw IllegalStateException("Missing flow factory in Realm configuration.")
+        else -> throw IllegalStateException("Wrong type of Realm.")
     }
 }
+
+// TODO figure out if we want to do this as a separate method or merge both in one that delivers changesets and results
```

Comment:
After thinking a bit about this,  I think we should introduce this method as well. IMO the one just returning the `RealmResults` are far easier to consume, this one is slightly annoying as you have to unpack things everytime, but it still has a lot of uses.